src/components/NotesView.vue

```diff
@@ -0,0 +1,270 @@
+		NcAppContentDetails,
+		NcButton,
+		Note,
+		NotesList,
+		NotesCaption,
+	},
+
+	directives: {
+		'observe-visibility': ObserveVisibility,
+	},
+
+	props: {
+		noteId: {
+			type: String,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			timeslots: [],
+			monthFormat: new Intl.DateTimeFormat(OC.getLanguage(), { month: 'long', year: 'numeric' }),
+			lastYear: new Date(new Date().getFullYear() - 1, 0),
+			showFirstNotesOnly: true,
+			showNote: true,
+		}
+	},
+
+	computed: {
+		getFilteredTotalCount() {
+			return store.getters.getFilteredTotalCount()
+		},
+
+		category() {
+			return store.getters.getSelectedCategory()
+		},
+
+		filteredNotes() {
+			return store.getters.getFilteredNotes()
+		},
+
+		displayedNotes() {
+			if (this.filteredNotes.length > 40 && this.showFirstNotesOnly) {
+				return this.filteredNotes.slice(0, 30)
+			} else {
+				return this.filteredNotes
+			}
+		},
+
+		// group notes by time ("All notes") or by category (if category chosen)
+		groupedNotes() {
+			if (this.category === null) {
+				return this.displayedNotes.reduce((g, note) => {
+					const timeslot = this.getTimeslotFromNote(note)
+					if (g.length === 0 || g[g.length - 1].timeslot !== timeslot) {
+						g.push({ timeslot, notes: [] })
+					}
+					g[g.length - 1].notes.push(note)
+					return g
+				}, [])
+			} else {
+				return this.displayedNotes.reduce((g, note) => {
+					if (g.length === 0 || g[g.length - 1].category !== note.category) {
+						g.push({ category: note.category, notes: [] })
+					}
+					g[g.length - 1].notes.push(note)
+					return g
+				}, [])
+			}
+		},
+		searchText: {
+			get() {
+				return store.state.searchText
+			},
+			set(value) {
+				store.commit('updateSearchText', value)
+			},
+		},
+	},
+
+	watch: {
+		category() { this.showFirstNotesOnly = true },
+	},
+
+	created() {
+		this.updateTimeslots()
+		setInterval(this.updateTimeslots, 1000 * 60)
+	},
+
+	methods: {
+		updateTimeslots() {
+			const now = new Date()
+			// define the time groups we want to allow
+			this.timeslots = [
+				{ t: new Date(now.getFullYear(), now.getMonth(), now.getDate()), l: t('notes', 'Today') },
+				{ t: new Date(now.getFullYear(), now.getMonth(), now.getDate() - 1), l: t('notes', 'Yesterday') },
+				{ t: new Date(now.getFullYear(), now.getMonth(), now.getDate() - now.getDay()), l: t('notes', 'This week') },
+				{ t: new Date(now.getFullYear(), now.getMonth(), now.getDate() - now.getDay() - 7), l: t('notes', 'Last week') },
+				{ t: new Date(now.getFullYear(), now.getMonth(), 1), l: t('notes', 'This month') },
+				{ t: new Date(now.getFullYear(), now.getMonth() - 1, 1), l: t('notes', 'Last month') },
+			]
+		},
+
+		categoryTitle(category) {
+			return categoryLabel(category)
+		},
+
+		categoryToLabel(category) {
+			return categoryLabel(category.substring(this.category.length + 1))
+		},
+
+		getTimeslotFromNote(note) {
+			if (note.favorite) {
+				return ''
+			}
+			const t = note.modified * 1000
```

Comment:
May move `1000` to const variable

Reply:
I'd still keep this as this is a pattern that js developers likely know well and a constant (or timestampToMilliseconds helper that I was thinking about) would be easy to be overseen in new code introduced and then we'd have two ways of doing that.